### PR TITLE
Apply StyleCop to several projects

### PR DESCRIPTION
**Pull request by torbacz**
## Description

- Apply StyleCop settings to 9 projects.

## Motivation and Context

- Partially resolves nanoFramework/Home#1101

## How Has This Been Tested?

- Local build.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by torbacz** [comment]
@Ellerbach 
I think we should get rid of SA1407 rule. Inserting brackets can cause bugs in code and also it's decreasing readability.
SA1407 : CSharp.Maintainability : Insert parentheses within the arithmetic expression to declare the operator precedence.

**Comment by Ellerbach** [comment]
> I think we should get rid of SA1407 rule. Inserting brackets can cause bugs in code and also it's decreasing readability.
> SA1407 : CSharp.Maintainability : Insert parentheses within the arithmetic expression to declare the operator precedence.

Can you give me an example? So we can compare with and without the rule. I have to be honest, I don't know them all by heart :-D

File: devices/Bmxx80/Bme280.cs [hunk 1]
```diff
@@ -141,10 +141,10 @@ private RelativeHumidity CompensateHumidity(int adcHumidity)
         {
             // The humidity is calculated using the compensation formula in the BME280 datasheet.
             double varH = TemperatureFine - 76800.0;
-            varH = (adcHumidity - (_bme280Calibration.DigH4 * 64.0 + _bme280Calibration.DigH5 / 16384.0 * varH)) *
-                (_bme280Calibration.DigH2 / 65536.0 * (1.0 + _bme280Calibration.DigH6 / 67108864.0 * varH *
-                                               (1.0 + _bme280Calibration.DigH3 / 67108864.0 * varH)));
-            varH *= 1.0 - _bme280Calibration.DigH1 * varH / 524288.0;
+            varH = (adcHumidity - ((_bme280Calibration.DigH4 * 64.0) + ((_bme280Calibration.DigH5 / 16384.0) * varH))) *
```
Comment (torbacz) [inline devices/Bmxx80/Bme280.cs:144]: @Ellerbach 
SA1407 example here.

Comment (Ellerbach) [reply devices/Bmxx80/Bme280.cs:144]: ok, case by case to release the rule. Now, that said, it's a rule that brings clarity and avoid mistakes. But I get your point. So make it the best with flexibility.

Comment (torbacz) [reply devices/Bmxx80/Bme280.cs:144]: Sorry, but I think i didn't get you. Should I remove the rule or not?

File: devices/Bh1750fvi/Bh1750fvi.cs [hunk 2]
```diff
@@ -10,19 +10,19 @@
...
 {
     /// <summary>
-    /// Ambient Light Sensor BH1750FVI
+    /// Ambient Light Sensor BH1750FVI.
     /// </summary>
     [Interface("Ambient Light Sensor BH1750FVI")]
     public class Bh1750fvi : IDisposable
     {
-        private const byte DefaultLightTransmittance = 0b_0100_0101;
+        private const byte DefaultLightTransmittance = 0b01000101;
```
Comment (rkalwak) [inline devices/Bh1750fvi/Bh1750fvi.cs:18]: From my experience problem is the second underscore. Try with 0b_10000101

File: devices/Ccs811/Error.cs [hunk 3]
```diff
@@ -6,33 +6,45 @@
...
+        /// <summary>No error.</summary>
         NoError = 0,
 
-        /// <summary>The CCS811 received an I²C write request addressed to this station but with
-        /// invalid register address ID</summary>
+        /// <summary>
+        /// The CCS811 received an I²C write request addressed to this station but with
+        /// invalid register address ID.
+        /// </summary>
         WriteRegisterInvalid = 0b0000_0001,
```
Comment (rkalwak) [inline devices/Ccs811/Error.cs:21]: Incosistent with other places - see my previous comment.

File: devices/RadioReceiver/Devices/Tea5767/FrequencyRange.cs [hunk 4]
```diff
@@ -9,12 +9,12 @@ namespace Iot.Device.RadioReceiver
     public enum FrequencyRange
     {
         /// <summary>
-        /// 76MHz - 90MHz
+        /// 76MHz - 90MHz.
```
Comment (rkalwak) [inline devices/RadioReceiver/Devices/Tea5767/FrequencyRange.cs:12]: Incosistent with other places - please check how you handled it here: devices/Bmxx80/StandbyTime.cs

File: devices/RadioReceiver/Devices/Tea5767/Tea5767.cs [hunk 5]
```diff
@@ -18,22 +18,23 @@ public class Tea5767 : RadioReceiverBase
...
         private I2cDevice _i2cDevice;
+
         // TEA5767 has no register. Setting TEA5767 requires sending 5 bytes of commands to it.
         // Default values: disable mute, XTAL is 13MHz, disable standby, enable SNC
-        private byte[] _registers = new byte[5] { 0, 0, 0b_0001_0000, 0b_0001_0010, 0 };
+        private byte[] _registers = new byte[5] { 0, 0, 0b00010000, 0b00010010, 0 };
 
         /// <summary>
-        /// TEA5767 mute.
+        /// Gets or sets a value indicating whether TEA5767 is muteed.
```
Comment (rkalwak) [inline devices/RadioReceiver/Devices/Tea5767/Tea5767.cs:27]:
```suggestion
        /// Gets or sets a value indicating whether TEA5767 is muted.
```

File: devices/RadioReceiver/Devices/Tea5767/Tea5767.cs [hunk 6]
```diff
@@ -18,22 +18,23 @@ public class Tea5767 : RadioReceiverBase
         public const byte DefaultI2cAddress = 0x60;
 
         private I2cDevice _i2cDevice;
+
         // TEA5767 has no register. Setting TEA5767 requires sending 5 bytes of commands to it.
         // Default values: disable mute, XTAL is 13MHz, disable standby, enable SNC
-        private byte[] _registers = new byte[5] { 0, 0, 0b_0001_0000, 0b_0001_0010, 0 };
+        private byte[] _registers = new byte[5] { 0, 0, 0b00010000, 0b00010010, 0 };
```
Comment (rkalwak) [inline devices/RadioReceiver/Devices/Tea5767/Tea5767.cs:24]: Incosistent with other places

File: devices/RadioTransmitter/Devices/Kt0803/Kt0803.cs [hunk 7]
```diff
@@ -117,19 +119,19 @@ private void SetFrequency(double frequency)
 
             // 3 bytes
             freq = (int)(frequency * 20);
-            freq &= 0b_1111_1111_1111;
+            freq &= 0b111111111111;
```
Comment (rkalwak) [inline devices/RadioTransmitter/Devices/Kt0803/Kt0803.cs:122]: Incosistent with other places

**Comment by rkalwak** [comment]
I would say that we either go with 0b11110001 format everywhere or with underscore.

**Comment by Ellerbach** [comment]
> I would say that we either go with 0b11110001 format everywhere or with underscore.

0b1111_0001 is the best format because it's easy for the eye to count how many elements are on the left and right side. We've been trying to be consistent in the initial .NET IoT bindings but seems that it's not the case everywhere. So do the best to fix those you can @torbacz !

File: devices/Bh1750fvi/Bh1750fvi.cs (hunk 2 above)
Comment (torbacz) [reply devices/Bh1750fvi/Bh1750fvi.cs:18]: Actually, the problem was in first one.
0b_10000101 -> invalid
0b1000_0101 -> valid

File: devices/RadioReceiver/Devices/Tea5767/FrequencyRange.cs (hunk 4 above)
Comment (torbacz) [reply devices/RadioReceiver/Devices/Tea5767/FrequencyRange.cs:12]: What do you mean?

**Comment by torbacz** [comment]
@rkalwak Thanks for comments! I think I fixed them all.
@Ellerbach I've went with 0b1111_0001 every where. Should be fixed in all changed files.

**Comment by Ellerbach** [comment]
@torbacz maybe a good time for a merge on those bindings? There are already quite a lot. Just pulled the main changes and let's see what's missing to merge those progresses.

**Comment by torbacz** [comment]
@Ellerbach Yep, feel free to merge it. I'm not planning to push any more right now.

**Review by Ellerbach — APPROVED** [review]
Thanks for all the improvements!

**Comment by Ellerbach** [comment]
> @Ellerbach Yep, feel free to merge it. I'm not planning to push any more right now.

So all merged :-) Thanks again!

**Pull request merged**